realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -883,13 +930,13 @@ void BaseRealSenseNode::publishFrame(
     ROS_DEBUG("publishFrame(...)");
     unsigned int width = 0;
     unsigned int height = 0;
-    unsigned int bpp = 1;
+    auto stream_format = f.get_profile().format();
     if (f.is<rs2::video_frame>())
     {
         auto timage = f.as<rs2::video_frame>();
         width = timage.get_width();
         height = timage.get_height();
-        bpp = timage.get_bytes_per_pixel();
+        stream_format = timage.get_profile().format();
```

Comment:
Why will this produce different results than line 933?

Reply:
I think I tested this with some flow, will verify this again tomorrow.

Reply:
Fixed